v2/yarpcprotobuf/protoc-gen-yarpc-go/integration_test.go

```diff
@@ -55,32 +63,57 @@ func newKeyValueClient(t *testing.T) keyvaluepb.StoreYARPCClient {
 }
 
 func newHelloClient(t *testing.T) streampb.HelloYARPCClient {
+	dialer := &yarpcgrpc.Dialer{}
+	require.NoError(t, dialer.Start(context.Background()))
+
+	outbound := &yarpcgrpc.Outbound{
+		URL:    &url.URL{Scheme: "http", Host: "127.0.0.1:8888"},
+		Dialer: dialer,
+	}
 	return streampb.NewHelloYARPCClient(yarpc.Client{
 		Caller:  "test",
 		Service: "hello",
+		Stream:  outbound,
 	})
 }
 
-func startProcedures(t *testing.T, service string, procedures []yarpc.Procedure) (stop func() error) {
+func startProcedures(t *testing.T, transport, service string, procedures []yarpc.Procedure) (stop func()) {
 	router := yarpcrouter.NewMapRouter(service)
 	router.Register(procedures)
 
-	inbound := &yarpchttp.Inbound{
-		Addr:   ":8888",
-		Router: router,
+	var inbound testInbound
+	switch transport {
+	case "http":
+		inbound = &yarpchttp.Inbound{
+			Addr:   ":8888",
```

Comment:
let's use a `listener` on port 0 instead of a static port so we don't have any port collisions between multiple tests.
This function could return the address and thread that through `newHelloClient`